conftest.py

```diff
@@ -1,20 +1,15 @@
+import os
+import tempfile
+
 import pytest
 
-@pytest.fixture(autouse=True)
-def _docdir(request):
-    """
-    Make sure that doctests run in a temporary directory so that any files that
-    are created as part of the test get removed automatically.
-    """
-    # Trigger ONLY for doctestplus.
-    try:
-        doctest_plugin = request.config.pluginmanager.getplugin("doctestplus")
-        if isinstance(request.node.parent, doctest_plugin._doctest_textfile_item_cls):
-            tmpdir = request.getfixturevalue('tmpdir')
-            with tmpdir.as_cwd():
-                yield
-        else:
+
+@pytest.fixture(scope="session", autouse=True)
+def temp_cwd():
+    original_cwd = os.getcwd()
+    with tempfile.TemporaryDirectory() as tmp:
```

Comment:
By using Python's `tempfile` and not the built-in feature of Pytest, how does this respond to

```
pytest --basetemp=/tmp/scratch
```

in directing where the tempdirs get stored?

Generally, the Pytest temp dirs do not get deleted when the Python process gets completed, so that the results of the test can be examined, and the Windows issue probably avoided.

Reply:
There's no impact on files stored in a pytest temp directory, only to files that are saved relative to the current working directory.  We don't do that in the regular test suite because it would leave junk behind, but it's hard to avoid in the docstring tests where we want the examples to work when copied and pasted into the user's console.

Reply:
(we chatted about this on Slack and I learned about the tmpdir_factory fixture)